src/ui/pages/SignTypedData/SignTypedData.tsx

```diff
@@ -337,34 +339,42 @@ function TypedDataDefaultView({
               >
                 Cancel
               </Button>
-
-              <SignMessageButton
-                wallet={wallet}
-                ref={signMsgBtnRef}
-                onClick={() => {
-                  if (!seenSigningData) {
-                    scrollSigningData();
-                  } else {
-                    signTypedData_v4();
+              {preferences ? (
```

Comment:
I would maybe bail out of rendering the whole component until we receive preferences

Reply:
This is a possible solution. Ideally, this preference fetch should be inside the component to avoid same code in multiple places.